hashicorp/dev-portal
# feat: adds custom icon card grid item support

## 🔗 Relevant links



- [Preview link](https://dev-portal-git-ksadds-custom-docs-landing-icon-tiles-hashicorp.vercel.app/hcp/docs) 🔎
- [Asana task](https://app.asana.com/0/1202702246158644/1202932567024650) 🎟️

## 🗒️ What

This PR introduces a new `iconCardGridItems` property to the docs landing template. If this is passed in the content file, it will overrides the root docs paths as the source for generating the icon tile grid. 

This was used in the HCP designs and I do think it adds good context in terms of UX for hcp. Definitely open to alternative approaches here and a discussion as to whether we want to use this for other products, as I know it's been asked for. 

## 📸 Design Screenshots

<img width="1036" alt="Screen Shot 2022-09-08 at 4 16 28 PM" src="https://user-images.githubusercontent.com/36613477/189241954-98d1da9e-3942-4c05-a8b5-12391ea38a5a.png">


## 🧪 Testing

- [Visit this link](https://dev-portal-git-ksadds-custom-docs-landing-icon-tiles-hashicorp.vercel.app/hcp/docs)
- Verify the custom links render and populate the tiles
- Visit another [product docs landing](https://dev-portal-git-ksadds-custom-docs-landing-icon-tiles-hashicorp.vercel.app/nomad/docs), verify the root docs paths still generate the tiles. 

## 💭 Anything else?

The designs show a 2-up grid, but imo that isn't worth the effort here.
